packages/wallets/src/features/cfd/screens/MT5TradeScreen/MT5TradeLink/MT5TradeLink.tsx

```diff
@@ -1,47 +1,121 @@
+import LinuxIcon from '../../../../../public/images/ic-linux-logo.svg';
+import MacOSIcon from '../../../../../public/images/ic-macos-logo.svg';
 import MT5Icon from '../../../../../public/images/ic-mt5.svg';
 import WindowsIcon from '../../../../../public/images/ic-windows-logo.svg';
-import MacOSIcon from '../../../../../public/images/ic-macos-logo.svg';
-import LinuxIcon from '../../../../../public/images/ic-linux-logo.svg';
+import { THooks, TPlatforms } from '../../../../../types';
+import { PlatformToTitleMapper } from '../../../constants';
 import './MT5TradeLink.scss';
 
 const AppToContentMapper = {
+    ctrader: {
+        icon: <WindowsIcon />,
+        link: 'https://getctrader.com/deriv/ctrader-deriv-setup.exe',
+        text: 'Download',
+        title: 'CTrader Windows App',
+    },
     linux: {
         icon: <LinuxIcon />,
+        link: 'https://www.metatrader5.com/en/terminal/help/start_advanced/install_linux',
         text: 'Learn more',
         title: 'MetaTrader 5 Linux app',
     },
     macos: {
         icon: <MacOSIcon />,
+        link: 'https://download.mql5.com/cdn/web/metaquotes.software.corp/mt5/MetaTrader5.dmg',
         text: 'Download',
         title: 'MetaTrader 5 MacOS app',
     },
     web: {
         icon: <MT5Icon />,
+        link: '',
         text: 'Open',
         title: 'MetaTrader 5 web',
     },
     windows: {
         icon: <WindowsIcon />,
+        link: 'https://download.mql5.com/cdn/web/deriv.com.limited/mt5/deriv5setup.exe',
         text: 'Download',
         title: 'MetaTrader 5 Windows app',
     },
 };
 
+const PlatformToLabelIconMapper = {
+    ctrader: <CTraderLabelIcon />,
+    derivez: <DerivEZLabelIcon />,
+    dxtrade: <DerivXLabelIcon />,
+};
+
 type TMT5TradeLinkProps = {
-    app: keyof typeof AppToContentMapper;
+    app?: keyof typeof AppToContentMapper;
+    isDemo?: THooks.ActiveWalletAccount['is_virtual'];
+    platform?: TPlatforms.All;
+    webtraderUrl?: THooks.MT5AccountsList['webtrader_url'];
 };
 
-const MT5TradeLink: FC<TMT5TradeLinkProps> = ({ app }) => {
+const MT5TradeLink: FC<TMT5TradeLinkProps> = ({ app = 'linux', isDemo = false, platform, webtraderUrl = '' }) => {
     const content = AppToContentMapper[app];
+    const { data: ctraderToken } = useCtraderServiceToken();
+    const { data: derivezToken } = useDerivezServiceToken();
+
+    const onClickWebTerminal = () => {
+        const { isStaging, isTestLink } = getPlatformFromUrl();
+        let url;
+        switch (platform) {
+            case 'dxtrade':
+                url = isDemo ? 'https://dx-demo.deriv.com' : 'https://dx.deriv.com';
+                break;
+            case 'ctrader':
+                url = isTestLink || isStaging ? 'https://ct-uat.deriv.com' : 'https://ct.deriv.com';
+                if (ctraderToken) url += `?token=${ctraderToken}`;
+                break;
+            case 'derivez':
+                url = 'https://dqwsqxuu0r6t9.cloudfront.net';
```

Comment:
is this correct url?
not to mention that it could potentially change with infrastructure change,

Reply:
yes, I took these links from `packages/cfd/src/Helpers/constants.ts` 
<img width="863" alt="Screenshot 2023-11-02 at 12 14 23 PM" src="https://github.com/binary-com/deriv-app/assets/103016120/851c1bb0-801b-4216-ae3a-5b8e1fe60851">

Reply:
However at the moment these links are only used inside the trade modal and not anywhere else, so if the link changes we can just change it in `MT5TradeLink.tsx`, otherwise if we are reusing this anywhere else in the future we can move these constants into the `constants` file in wallets

Reply:
Thanks @adrienne-deriv. I understand we don't want to import it from the cfd "package", as we're trying to remove the dependencies between the "packages".

But would it make sense to create a constants or config file in `wallets` to put this?

Reply:
We have a `constants` file in wallets, but its only constants that we are commonly reusing